drivers/Watchdog.h

```diff
@@ -20,21 +20,22 @@
 
 #ifdef DEVICE_WATCHDOG
 
-#include "watchdog_api.h"
-
 #include <cstdio>
-
-
+#include "mbed_error.h"
+#include "platform/mbed_critical.h"
+#include "platform/mbed_power_mgmt.h"
+#include "mbed_assert.h"
 namespace mbed {
+
 /** \addtogroup drivers */
 /** A system timer that will reset the system in the case of system failures or
  *  malfunctions.
  *
```

Comment:
Would be good to explain that this is not driving the HW watchdog directly but can be instantiated multiple times/etc